lib/ansible/modules/source_control/github_release.py

```diff
@@ -30,23 +30,55 @@
     - Fetch metadata about Github Releases
 version_added: 2.2
 options:
-    token:
-        required: true
-        description:
-            - Github Personal Access Token for authenticating
-    user:
-        required: true
-        description:
-            - The GitHub account that owns the repository
-    repo:
-        required: true
-        description:
-            - Repository name
-    action:
-        required: true
-        description:
-            - Action to perform
-        choices: [ 'latest_release' ]
+  token:
+    required: true
+    description:
+      - Github Personal Access Token for authenticating
+  user:
+    required: true
+    description:
+      - The GitHub account that owns the repository
+  repo:
+    required: true
+    description:
+      - Repository name
+  action:
+    required: true
+    description:
+      - Action to perform
+    choices: [ 'latest_release', 'create_release' ]
+  tag:
+    required: false
+    description:
+      - Tag name when creating a release. Required when using create_release.
+    version_added: 2.4
+  target:
+    required: false
+    description:
+      - Target of release when creating a release
+    version_added: 2.4
+  name:
+    required: false
+    description:
+      - Name of release when creating a release
+    version_added: 2.4
+  body:
+    required: false
+    description:
+      - Description of the release when creating a release
+    version_added: 2.4
+  draft:
+    required: false
+    description:
+      - Sets if the release is a draft or not. (boolean)
+    default: false
+    version_added: 2.4
+  prerelease:
+    required: false
+    description:
+      - Sets if the release is a prerelease or not. (boolean)
+    default: falseA
+    version_added: 2.4
```

Comment:
There is a `upload_asset` method on the `release` object. So it should be easy to add this.